tracee-ebpf/main.go

```diff
@@ -114,12 +116,30 @@ func main() {
 				fmt.Fprintf(os.Stderr, "warning: could not check enabled kconfig features, trying to continue anyway\n")
 			}
 
-			// try to discover distro by /etc/os-release, fallback to kernel version
+			// OS release information
+
+			osReleaseFilePath, err := checkEnvPath("TRACEE_OSRELEASE_FILE") // useful if users wants to mount host os-release in a container
+			if err != nil {
+				return err
+			} else if osReleaseFilePath == "" {
+				osReleaseFilePath = "/etc/os-release"
+			}
 
-			btfinfo := helpers.NewBTFInfo()
-			if err = btfinfo.DiscoverDistro(); err != nil {
+			OSInfo, err := helpers.NewOSInfo(osReleaseFilePath)
+			if err != nil {
+				OSInfo, err = helpers.NewOSInfoRelease() // at least check $(uname -r) if os-release could not be found
+				if err != nil {
+					return err
+				}
```

Comment:
instead of exposing the internal information source to the user, abstract it away and put the decision logic in the implementation